psychopy/visual/dot.py

```diff
@@ -504,3 +504,199 @@ def _update_dotsXY(self):
+            self._dotsLife[dead] = self.dotLife
+        else:
+            dead = numpy.zeros(self.nDots, dtype=bool)
+
+        
+        # update dead dots
+        if sum(dead):
+            self._verticesBase[dead, :] = self._newDotsXY(sum(dead))
+            self._dotsDir[dead] = numpy.random.vonmises(self.dir, self.kappa, sum(dead))
+
+
+        cosDots = numpy.cos(self._dotsDir)
+        sinDots = numpy.sin(self._dotsDir)
+
+        self._verticesBase[:, 0] += self.speed * cosDots
+        self._verticesBase[:, 1] += self.speed * sinDots
+
+        # find Out-of-bounds dots
+        if self.fieldShape in (None, 'square', 'sqr'):
+            out0 = (numpy.abs(self._verticesBase[:, 0]) > 0.5*self.fieldSize[0])
+            out1 = (numpy.abs(self._verticesBase[:, 1]) > 0.5*self.fieldSize[1])
+            outofbounds = out0 + out1
+
+        elif self.fieldShape == 'circle':
+            normXY = self._verticesBase / 0.5 / self.fieldSize
+            outofbounds = (numpy.hypot(normXY[:, 0], normXY[:, 1]) > 1)   
+
+        # update any out of bound dots dots
+        if sum(outofbounds):
+            self._verticesBase[outofbounds, :] = self._newDotsXY(sum(outofbounds))
+            self._dotsDir[outofbounds] = numpy.random.vonmises(self.dir, self.kappa, sum(outofbounds))
+
+        # update the pixel XY coordinates in pixels (using _BaseVisual class)
+        self._updateVertices()
+
+
+class GoldShadlenDotStim(DotStim):
+
+    ''' This version of the Random Dot Motion-task is the one first described
+    in Gold & Shadlen (2003). It is very similar to the standard random dot
+    kinematogram of Britten et al. (1992), but uses 3 separate frames of dots
+    that are subsequently projected. This version of the task has been used in many
+    model-based fMRI-studies on perception, like Forstmann et al., (2008) and
+    Mulder et al. (2012).
+
+     * Gold, J. I., & Shadlen, M. N. (2003). The influence of behavioral context on
+     the representation of a perceptual decision in developing oculomotor commands.
+     The Journal of Neuroscience, 23(2), 632–651.
+     * Forstmann, B. U., Dutilh, G., Brown, S., Neumann, J., Cramon, Von, D. Y.,
+     Ridderinkhof, K. R., & Wagenmakers, E.-J. (2008). Striatum and pre-SMA facilitate
+     decision-making under time pressure. Proc Natl Acad Sci USA, 105(45), 17538–17542.
+     * van Maanen, L., Brown, S. D., Eichele, T., Wagenmakers, E.-J., Ho, T.,
+     Serences, J., & Forstmann, B. U. (2011). Neural correlates of trial-to-trial
+     fluctuations in response caution. The Journal of Neuroscience, 31(48), 17488–17495.
+     http://doi.org/10.1523/JNEUROSCI.2924-11.2011
+     '''
+
+    def __init__(self, 
+                 win,
+                 nFrames=3,
+                 signalDots='different',
+                 noiseDots='walk',
+                 fieldShape='circle',
+                 dotLife=0,
+                 nDots=500,
+                 *args, **kwargs):
```

Comment:
Please document all parameters, and explicitly list parameters that are going to be passed to the parent class.